docs/core/seeding.rst

```diff
@@ -288,19 +358,20 @@ This function assigns a weight (which should correspond to the likelihood that
 a seed is good) to all seeds and calls the detector specific cuts to apply a
 hard cut or modify the weight. The weight is a “soft cut”, in that it is only
 used to discard tracks if many seeds are created for the same middle SP in
-SeedFilter::filterSeeds_1SpFixed
+SeedFilter::filterSeeds_1SpFixed. This process is important to improving computational
+performance and the quality of the final track collections by rejecting lower-quality seeds.
 
-The weight is influenced by:
+The weight can be influenced by:
 
-#. The impact parameter (the higher the distance the worse)
-#. The number of seeds which may belong to the same particle track
+#. The transverse (`d_{0}`) and longitudinal (`z_{0}`) impact parameters (the higher the distance the worse)
+#. The number of seeds which may belong to the same particle track (`N_{t}`)
 #. Optional detector specific cuts.
 
-
-The impact parameter is multiplied by the configured factor and subtracted from
+The transverse impact parameter is multiplied by the configured factor and subtracted from
 the weight, as seeds with higher impact parameters are assumed to be less
 likely to stem from a particle than another seed using the same middle SP with
-smaller impact parameters.
+smaller impact parameters. The longitudinal impact parameter is subtracted from
+the weight if configured.
 
 The number of seeds only differing in top SP which have similar helix radius
 and the same sign (i.e. the same charge) is used to increase the weight, as it
```

Comment:
text below introduces "measurement". I suggest to stay with SP. In any case does this cover SPs in the overlapping modules? The logic below is quite hard to follow.